src/main/java/org/jenkinsci/plugins/testresultsanalyzer/TestResultsAnalyzerAction.java

```diff
@@ -106,18 +110,32 @@ private JSONArray getBuildsArray(List<Integer> buildList) {
 		return jsonArray;
 	}
 
-	private List<Integer> getBuildList(int noOfBuilds) {
-		if ((noOfBuilds <= 0) || (noOfBuilds >= builds.size())) {
-			return builds;
+	private List<Integer> getBuildList(int noOfBuilds, String showCompileFail) {
+		boolean showFail = showCompileFail.equals("show");
+
+		if ((noOfBuilds <= 0) || (noOfBuilds > builds.size())) {
+			noOfBuilds = builds.size();
 		}
 		List<Integer> buildList = new ArrayList<Integer>();
-		for (int i = (noOfBuilds - 1); i >= 0; i--) {
-			buildList.add(builds.get(i));
+		for(int i = 0; i < noOfBuilds; i++) {
+			int index = builds.get(i);
+
+			if(showFail || !(compileFailedBuilds.contains(index))) {
+				buildList.add(index);
+			}
 		}
-		Collections.reverse(buildList);
```

Comment:
Please revert this change. As during showing selective builds without this change the build is shown in ascending order rather then descending order.